client/my-sites/themes/theme-showcase.jsx

```diff
@@ -58,6 +60,98 @@ const optionShape = PropTypes.shape( {
 	action: PropTypes.func,
 } );
 
+const bannerShape = {
+	siteId: PropTypes.number,
+};
+
+const bannerClickHandler = ( themeId, tracksEvent, { siteId, getBannerUrl, getCurrentTheme } ) => {
+	const url = getBannerUrl( themeId, siteId );
+	const tracksData = {
+		site_id: siteId,
+		current_theme: getCurrentTheme( siteId ),
+		timestamp: new Date().toGMTString(),
+	};
+	recordTracksEvent( tracksEvent, tracksData );
+	page( url );
+};
+
+class PhotoBlogBanner extends PureComponent {
+	static propTypes = bannerShape;
+
+	render() {
+		const onClick = () => {
+			bannerClickHandler( 'photo-blog', 'calypso_showcase_banner_photo_blog_click', this.props );
+		};
+		return (
+			<ThemesBanner
+				title={ translate( 'Are you a photographer? An artist?' ) }
+				description={ translate(
+					'Explore {{b}}PHOTO BLOG{{/b}}, an elegant theme designed to showcase your visual masterpieces.',
+					{
+						components: {
+							b: <strong />,
+						},
+					}
+				) }
+				action={ onClick }
+				actionLabel={ translate( 'See the theme' ) }
+				backgroundColor={ '#3FE6AF' }
+				image={ '/calypso/images/themes-banner/photo-blog.png' }
+				imageTransform={ 'translateY(-4.4%) translateX(17%)' }
+				imageAttrs={ {
+					alt: translate( 'Photo Blog Theme' ),
+					width: 390,
+				} }
+			/>
+		);
+	}
+}
+
+class SmallBusinessBanner extends PureComponent {
+	static propTypes = bannerShape;
+
+	render() {
+		const onClick = () => {
+			bannerClickHandler(
+				'small-business',
+				'calypso_showcase_banner_small_business_click',
+				this.props
+			);
+		};
+		return (
+			<ThemesBanner
+				title={ translate( 'Do you run a small business?' ) }
+				description={ translate(
+					'We know you’re crunched for time. We created the quick-setup {{b}}SMALL BUSINESS{{/b}} theme just for you.',
+					{
+						components: {
+							b: <strong />,
+						},
+					}
+				) }
+				action={ onClick }
+				actionLabel={ translate( 'See the theme' ) }
+				backgroundColor={ '#3d596d' }
+				image={ '/calypso/images/themes-banner/small-business.png' }
+				imageTransform={ 'translateY(-19%) translateX(17%)' }
+				imageAttrs={ {
+					alt: translate( 'Small Business Theme' ),
+					width: 410,
+				} }
+			/>
+		);
+	}
+}
+
+const ShowcaseBanner = sample(
+	[ PhotoBlogBanner, SmallBusinessBanner ].map( component =>
+		connect( state => ( {
+			getBannerUrl: curry( getThemeDetailsUrl )( state ),
+			getCurrentTheme: curry( getActiveTheme )( state ),
+		} ) )( component )
+	)
+);
```

Comment:
Wow, quite complex 😅 Instead of doing the same thing for each individual banner, I'd much rather suggest doing it once to the `ThemesBanner` component that both are based on 😉 (right in `themes-banner/index.jsx`).

Reply:
(I'd suggest making `themeId` a `ThemesBanner` prop to that end.)

Reply:
Also, we generally try to avoid passing (bound) functions as props from `connect()` to the wrapped component...